engine/orchestration/src/org/apache/cloudstack/engine/orchestration/NetworkOrchestrator.java

```diff
@@ -671,8 +672,19 @@ public void doInTransactionWithoutResult(TransactionStatus status) {
                                 vpcId, offering.getRedundantRouter());
                         vo.setDisplayNetwork(isDisplayNetworkEnabled == null ? true : isDisplayNetworkEnabled);
                         vo.setStrechedL2Network(offering.getSupportsStrechedL2());
-                        networks.add(_networksDao.persist(vo, vo.getGuestType() == Network.GuestType.Isolated,
-                                finalizeServicesAndProvidersForNetwork(offering, plan.getPhysicalNetworkId())));
+                        NetworkVO networkPersisted = _networksDao.persist(vo, vo.getGuestType() == Network.GuestType.Isolated,
+                                finalizeServicesAndProvidersForNetwork(offering, plan.getPhysicalNetworkId()));
+                        networks.add(networkPersisted);
+
+                        if (predefined instanceof NetworkVO){
+                            try{
+                                NetworkGuruAdditionalFunctions functions = (NetworkGuruAdditionalFunctions) guru;
+                                functions.finalizeNetworkDesign(networkPersisted.getId(), ((NetworkVO)predefined).getVlanIdAsUUID());
+                            }
+                            catch (Exception e){
+                                //Couldn't cast to NetworkGuruAdditionalFunctions, continue
```

Comment:
The code should not catch exception because it is simply too broad.
Caught exceptions cannot be silently ignored (i.e. having an empty `catch` block).